src/gui/AutomatableModelView.cpp

```diff
@@ -258,7 +279,227 @@ void AutomatableModelViewSlots::removeConnection()
+	{
+		// selecting the track with the most amount of clips
+		// connected to this model
+		AutomationTrack* maxTrack = dynamic_cast<AutomationTrack*>((*clips)[0]->getTrack());
+		int maxTrackCount = 1;
+		for (size_t i = 1; i < clips->size(); i++)
+		{
+			int currentCount = 0;
+			for (size_t j = 0; j < clips->size(); j++)
+			{
+				if ((*clips)[i]->getTrack() == (*clips)[j]->getTrack())
+				{
+					currentCount++;
+				}
+			}
+			if (maxTrackCount < currentCount)
+			{
+				maxTrackCount = currentCount;
+				maxTrack = dynamic_cast<AutomationTrack*>((*clips)[i]->getTrack());;
+			}
+		}
+		output = maxTrack;
+	}
+	else if (canAddNewTrack == true)
+	{
+		// adding new track
+		output = new AutomationTrack(getGUI()->songEditor()->m_editor->model(), false);
+	}
+	return output;
+}
+AutomationClip* AutomatableModelViewSlots::getCurrentAutomationClip(AutomationTrack* track, bool canAddNewClip, bool searchAfter)
+{
+	AutomationClip* output = nullptr;
+	std::vector<Clip*> trackClips = track->getClips();
+	TimePos timePos = static_cast<TimePos>(Engine::getSong()->getPlayPos());
+	
 
+	bool tryAdding = false;
+	if (trackClips.size() > 0)
+	{
+		// getting the closest clip that start before or after the global time position
+		tick_t closestTime = -1;
+		Clip* closestClip = nullptr;
+		for (Clip* currentClip : trackClips)
+		{
+			tick_t currentTime = currentClip->startPosition().getTicks();
+			if ((searchAfter == false && (currentTime > closestTime || closestTime < 0) && timePos.getTicks() > currentTime)
+				|| (searchAfter == true && (currentTime < closestTime || closestTime < 0) && timePos.getTicks() <= currentTime))
+			{
+				closestTime = currentTime;
+				closestClip = currentClip;
+			}
+		}
+
+		// in some cases there could be no clips before or after the global time position
+		// if this is the case, try adding a new one
+		// (if this fails, return nullptr)
+		if (closestClip == nullptr)
+		{
+			tryAdding = true;
+		}
+		else
+		{
+			output = dynamic_cast<AutomationClip*>(closestClip);
+		}
+	}
+	else
+	{
+		tryAdding = true;
+	}
+	if (tryAdding == true && canAddNewClip == true)
+	{
+		// adding a new clip
+		output = makeNewClip(track, timePos, true);
+	}
+	return output;
+}
+const TimePos AutomatableModelViewSlots::getNearestAutomationNode(AutomationTrack* track, AutomationClip** clipOut)
+{
+	TimePos output;
+	AutomationClip* minClip = nullptr;
+	int minDistance = -1;
+
+	TimePos timePos = static_cast<TimePos>(Engine::getSong()->getPlayPos());
+	// getting the clips before and after the global time position
+	AutomationClip* clipBefore = getCurrentAutomationClip(track, false, false);
+	AutomationClip* clipAfter = getCurrentAutomationClip(track, false, true);
+
+	if (clipBefore != nullptr && clipBefore->hasAutomation() == true)
+	{
+		// getting nearest node
+		// in the clip that starts before this
+		for(AutomationClip::timeMap::const_iterator it = clipBefore->getTimeMap().begin(); it != clipBefore->getTimeMap().end(); ++it)
```

Comment:
Use auto here in for loop declaration

Reply:
Ideally you should use range based for loop here.

Reply:
To my understanding only iterators can be used here.